### ISSUE TITLE
numpy.allclose does not work as expected for unsigned dtypes

### ISSUE THREAD

**Issue by AngledLuffa** [numpy/numpy#5932]
Here is a brief example:

```
>>> aa = numpy.array([1], dtype=numpy.uint8)
>>> bb = numpy.array([2], dtype=numpy.uint8)
>>> numpy.allclose(aa, bb, atol=10)
False
>>> AA = numpy.array([1])
>>> BB = numpy.array([2])
>>> numpy.allclose(AA, BB, atol=10)
True
```

This is using version 1.9.1

**Comment by rgommers** [comment]
I can reproduce this, but only with Python 3. Which Python version are you using?

**Comment by AngledLuffa** [comment]
Good question, sorry for not indicating that.  It happens for me with Python 2.7.9 as well.

**Comment by argriffing** [comment]
FWIW I cannot replicate this on linux with recent numpy and python 2.7.
```
$ python
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> a = np.array([1], dtype=np.uint8)
>>> b = np.array([2], dtype=np.uint8)
>>> np.allclose(a, b, atol=10)
True
```

**Comment by AngledLuffa** [comment]
Ah, yes.  I can see it is fixed in the latest numpy on 2.7.  At least it is
a useful bug report for python 3 :)

**Comment by mattip** [comment]
Cannot replicate, even with cpython 3.6.6. Please open a new issue if the problem can be reproduced with a different example
